programs/perpetuals/tests/tests_suite/basic_interactions.rs

```diff
@@ -0,0 +1,245 @@
+                &eth_mint,
+                &keypairs[USER_MARTIN].pubkey(),
+                &keypairs[ROOT_AUTHORITY],
+                2_000_000_000,
+            )
+            .await;
+        }
+
+        // Paul, 150 USDC
+        {
+            utils::initialize_and_fund_token_account(
+                &mut program_test_ctx,
+                &usdc_mint,
+                &keypairs[USER_PAUL].pubkey(),
+                &keypairs[ROOT_AUTHORITY],
+                150_000_000,
+            )
+            .await;
+
+            utils::initialize_token_account(
+                &mut program_test_ctx,
+                &eth_mint,
+                &keypairs[USER_PAUL].pubkey(),
+            )
+            .await;
+        }
+    }
+
+    let (pool_pda, _, lp_token_mint_pda, _, _) = utils::setup_pool_with_custodies_and_liquidity(
+        &mut program_test_ctx,
+        &keypairs[MULTISIG_MEMBER_A],
+        "FOO",
+        &keypairs[PAYER],
+        multisig_signers,
+        vec![
+            utils::SetupCustodyWithLiquidityParams {
+                setup_custody_params: utils::SetupCustodyParams {
+                    mint: usdc_mint,
+                    decimals: 6,
+                    is_stable: true,
+                    target_ratio: 5_000,
+                    min_ratio: 0,
+                    max_ratio: 10_000,
+                    initial_price: 1_000_000,
+                    initial_conf: 10_000,
+                    pricing_params: None,
+                    permissions: None,
+                    fees: None,
+                },
+                // Alice add 1k USDC liquidity
+                liquidity_amount: 1_000_000_000,
+                payer: utils::copy_keypair(&keypairs[USER_ALICE]),
+            },
+            utils::SetupCustodyWithLiquidityParams {
+                setup_custody_params: utils::SetupCustodyParams {
+                    mint: eth_mint,
+                    decimals: 9,
+                    is_stable: false,
+                    target_ratio: 5_000,
+                    min_ratio: 0,
+                    max_ratio: 10_000,
+                    initial_price: 1_676_040_000_000,
+                    initial_conf: 10_000_000_000,
+                    pricing_params: None,
+                    permissions: None,
+                    fees: None,
+                },
+                // Martin add 1 ETH liquidity
+                liquidity_amount: 1_000_000_000,
+                payer: utils::copy_keypair(&keypairs[USER_MARTIN]),
+            },
+        ],
+    )
+    .await;
+
+    // Simple open/close position
+    {
+        // Martin: Open 50 USDC position
+        let position_pda = instructions::test_open_position(
+            &mut program_test_ctx,
+            &keypairs[USER_MARTIN],
+            &keypairs[PAYER],
+            &pool_pda,
+            &usdc_mint,
+            OpenPositionParams {
+                // max price paid (slippage implied)
+                price: 1_050_000,
```

Comment:
I think we could reuse the 
```

    fn scale(amount: u64, decimals: u8) -> u64 {
        math::checked_mul(amount, 10u64.pow(decimals as u32)).unwrap()
    }

    fn scale_f64(amount: f64, decimals: u8) -> u64 {
        math::checked_as_u64(
            math::checked_float_mul(amount, 10u64.pow(decimals as u32) as f64).unwrap(),
        )
        .unwrap()
    }
```

From unit tests, makes things more readable by using default values? wdyt

Reply:
would become something like scale(1.05, USDC_Decimals)

Reply:
I think so, even with a cfg(test) flag, @askibin ? It's duplicated in several places already, but only for tests purposes